process-document/src/test/kotlin/com/ritense/processdocument/service/ProcessDocumentKotlinServiceIntTest.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.processdocument.service
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.ritense.document.domain.Document
+import com.ritense.document.domain.impl.JsonSchemaDocumentId
+import com.ritense.document.domain.impl.request.NewDocumentRequest
+import com.ritense.document.service.DocumentService
+import com.ritense.processdocument.BaseIntegrationTest
+import com.ritense.processdocument.repository.ProcessDocumentInstanceRepository
+import com.ritense.valtimo.service.CamundaProcessService
+import org.camunda.bpm.engine.RuntimeService
+import org.camunda.bpm.engine.TaskService
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.transaction.annotation.Transactional
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+
+@Transactional
+class ProcessDocumentKotlinServiceIntTest: BaseIntegrationTest() {
+
+    @Autowired
+    lateinit var runtimeService: RuntimeService
+
+    @Autowired
+    lateinit var processDocumentInstanceRepository: ProcessDocumentInstanceRepository
+
+    @Autowired
+    lateinit var processDocumentAssociationService: ProcessDocumentAssociationService
+
+    @Autowired
+    lateinit var documentService: DocumentService
+
+    @Autowired
+    lateinit var taskService: TaskService
+
+    @Autowired
+    lateinit var objectMapper: ObjectMapper
+
+    @Autowired
+    lateinit var camundaProcessService: CamundaProcessService
+
+    lateinit var documentJson: String
+    lateinit var document: Document
+
+    @BeforeEach
+    fun init() {
+        documentJson =
+            """
+            {
+                "street": "aStreet",
+                "houseNumber": 1
+            }
+            """.trimIndent()
+    }
+
+    @Test
+    @Throws(JsonProcessingException::class)
+    fun `should start process by process definition key`() {
```

Comment:
Missing a test for a non existing process definition key.

Reply:
don't think there is a need for when a document does not exist because the business key from the parent process is mandatory and is the document id.